packages/salesforcedx-vscode-core/src/commands/forceOrgOpen.ts

```diff
@@ -6,24 +6,64 @@
  */
 
 import {
+  CliCommandExecutor,
   Command,
   SfdxCommandBuilder
 } from '@salesforce/salesforcedx-utils-vscode/out/src/cli';
+import { ContinueResponse } from '@salesforce/salesforcedx-utils-vscode/out/src/types';
+import * as vscode from 'vscode';
 import { nls } from '../messages';
+import { telemetryService } from '../telemetry';
+import { getRootWorkspacePath } from '../util';
 import {
   EmptyParametersGatherer,
   SfdxCommandlet,
   SfdxCommandletExecutor,
   SfdxWorkspaceChecker
 } from './commands';
 
+interface CLIOrgData {
+  status: number;
+  result: {
+    orgId: string;
+    url: string;
+    username: string;
+  };
+}
+
 class ForceOrgOpenExecutor extends SfdxCommandletExecutor<{}> {
   public build(data: {}): Command {
-    return new SfdxCommandBuilder()
+    const builder = new SfdxCommandBuilder()
       .withDescription(nls.localize('force_org_open_default_scratch_org_text'))
       .withArg('force:org:open')
-      .withLogName('force_org_open_default_scratch_org')
-      .build();
+      .withLogName('force_org_open_default_scratch_org');
+    if (process.env.SFDX_CONTAINER_MODE) {
+      builder.withArg('-r').withJson();
```

Comment:
`-r` URL only and `withJson` to parse the stdout form the CLI

Reply:
Let's use the long URLs. Not the short one.

Reply:
I don't think the URL is shortened, at least it looks quite verbose. The `-r` flag just prevents the CLI from opening the browser and only returns the org info in the output. I'm not aware of any differences in the URL it returns.

Reply:
Sorry, mistyped. I meant use `--url` instead of `-r`.

Reply:
ok that makes more sense